### PR TITLE
Loading product attributes on category page

### LINKED ISSUE #3220: Call the `attribute/list` for all the attributes put in the `entities.productListWithChildren.includeFields`
## What is the motivation for adding/enhancing this feature?

Related to: https://github.com/DivanteLtd/vue-storefront/issues/3199

We should pre-fetch the metadata of the attribute to all `includedFields` - currently, we just load the `defaultFilters` which could not be enough to properly display the other fields.

On the product page - there is another `attribute/list` call for all the product attributes anyway

## What are the acceptance criteria 

- [ ] the attributes meta data loaded for all the product fields used on the category page, not only for the default flters

## Can you complete this feature request by yourself?

- [ ] YES
- [ ] NO

## Which [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html) state this refers to? Info for developer.
Pick one option.

- [x] This is a normal feature request. This should be available on https://test.storefrontcloud.io and then after tests this can be added to next Vue Storefront version. In this case Developer should create branch from `develop` branch and create Pull Request `2. Feature / Improvement` back to `develop`.
- [ ] (**Pick this option only if you're sure**) This is an important improvement request for current Release Candidate version on https://next.storefrontcloud.io and should be placed in next RC version. In this case Developer should create branch from `release` branch and create Pull Request `3. Stabilisation fix` back to `release`.
- [ ] (**Pick this option only if you're sure**) This is a critical improvement request for current Stable version on https://demo.storefrontcloud.io and should be placed in next stable version. In this case Developer should create branch from `hotfix` or `master` branch and create Pull Request `4. Hotfix` back to `hotfix`.

### PR DESCRIPTION
**Pull request by andrzejewsky**
### Related issues

closes #3220

### Short description and why it's useful
It loads all of attributes mentioned in `config.entities.productListWithChildren.includeFields` on the category page.

### Which environment this relates to
- [x] Test version (https://test.storefrontcloud.io) - this is a new feature or improvement for Vue Storefront. I've created branch from `develop` branch and want to merge it back to `develop`
- [ ] RC version (https://next.storefrontcloud.io) - this is a stabilisation fix for Release Candidate of Vue Storefront. I've created branch from `release` branch and want to merge it back to `release`
- [ ] Stable version (https://demo.storefrontcloud.io) - this is an important fix for current stable version. I've created branch from `hotfix` or `master` branch and want to merge it back to `hotfix`

### Upgrade Notes and Changelog

- [x] No upgrade steps required (100% backward compatibility and no breaking changes)
- [ ] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/doc/Upgrade%20notes.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing VS sites with this new feature

### Contribution and currently important rules acceptance
- [x] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)
- [x] I read the [TypeScript Action Plan](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/TypeScript%20Action%20Plan.md) and adjusted my PR according to it
- [x] I read about [Vue Storefront Modules](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/api-modules/about-modules.md) and I am aware that every new feature should be a module

### CODE REVIEW

File: src/themes/default/pages/Category.vue [hunk 1]
```diff
@@ -84,12 +84,13 @@ import config from 'config'
...
 import ButtonFull from 'theme/components/theme/ButtonFull.vue'
 import { mapGetters } from 'vuex'
+import uniq from 'lodash-es/uniq'
 import onBottomScroll from '@vue-storefront/core/mixins/onBottomScroll'
 
 const composeInitialPageState = async (store, route) => {
   try {
     await store.dispatch('attribute/list', { // load filter attributes for this specific category
-      filterValues: config.products.defaultFilters, // TODO: assign specific filters/ attribute codes dynamicaly to specific categories
+      filterValues: uniq([...config.products.defaultFilters, ...config.entities.productListWithChildren.includeFields]), // TODO: assign specific filters/ attribute codes dynamicaly to specific categories
```
Comment (pkarw) [inline src/themes/default/pages/Category.vue]: To be on the safe side please add the `config.entities.productListChildren.includeFields` - as from 1.11 @patzick is using only this set of fields in the Category page. The rest is fine. Thanks!

Comment (andrzejewsky) [reply src/themes/default/pages/Category.vue]: @pkarw There already is the `config.entities.productListChildren.includeFields` - do you mean keep only this field and remove this `uniq` function call with `config.products.defaultFilters`?

Comment (pkarw) [reply src/themes/default/pages/Category.vue]: No, sorry I meant `config.entities.productList.includeFields` - doing uniq over `config.entities.productListChildren.includeFields` + `config.entities.productList.includeFields`.

However - I think that Patryk is currently using `config.entities.productListChildren.includeFields` (I doublechecked it) so this comment makes no sense.

**Comment by pkarw** [comment]
@andrzejewsky whats more do we need here in order to merge t/

**Comment by andrzejewsky** [comment]
@pkarw so basically that's it, the attributes are loading when you just go to the category page. Although... we can consider some kind of optimisation - maybe fetching metadata for all of attributes is not necessary for everyone and sometimes the `defaultFilters` are enough... so we can add some sort of the config option, but don't know if it's really demanded.

**Comment by pkarw** [comment]
Yes, this is true. This is related to #3281 (however this one is about the SSR mode and INITIAL_STATE). As a separate PR plesae try to limit the _INITIAL_STATE_ (#3281); for some shops it's really HUGE (like 700Kb and more)

**Pull request merged**